### PR TITLE
Updated changelog for v2.8.4, v2.8.5, v2.8.6, v2.9.0

### PR DESCRIPTION
**Pull request by kdomanski**
**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,158 @@
...
+ppropriate query parameter instead. [#2315](https://github.com/kubermatic/kubermatic/issues/2315) ([p0lyn0mial](https://github.com/p0lyn0mial))
+- [ACTION REQUIRED] kubermatic API command line args were changed: `oidc-issuer-url` to `oidc-url`, `oidc-client-id` to `oidc-authenticator-client-id`, `oidc-offline-access-as-scope` to `oidc-issuer-offline-access-as-scope`, `oidc-redirect-uri` to `oidc-issuer-redirect-uri` and new was added: `oidc-issuer-client-id` [#2338](https://github.com/kubermatic/kuberm
+atic/issues/2338) ([zreigz](https://github.com/zreigz))
+- Kubermatic now validates nodes synchronously [#2340](https://github.com/kubermatic/kubermatic/issues/2340) ([alvaroaleman](https://github.com/alvaroaleman))
+- Added endpoints to manage Machine Deployments [#2357](https://github.com/kubermatic/kubermatic/issues/2357) ([maciaszczykm](https://github.com/maciaszczykm))
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Replace that with "Kubermatic now manages Nodes as group via the NodeGroup feature"? No onw can use that endpoints and most of or customers wont even know what a `MachineDeployment` is

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,158 @@
...
+- Components will no longer be shown as as unhealthy when only some replicas are up [#2358](https://github.com/kubermatic/kubermatic/issues/2358) ([mrIncompetent](https://github.com/mrIncompetent))
+- Kubernetes API servers can now be used with OpenID authentication[ACTION REQUIRED] to enable the OpenID for kubernetes API server the users must set `-feature-gates=OpenIDConnectTokens=true` and provide `-oidc-issuer-url`, `-oidc-issuer-client-id` when running the controller. [#2370](https://github.com/kubermatic/kubermatic/issues/2370) ([zreigz](https://git
+hub.com/zreigz))
+- [ACTION REQUIRED] Resource limits for control plane containers have been increased. This might require additional resources for the seed cluster [#2395](https://github.com/kubermatic/kubermatic/issues/2395) ([mrIncompetent](https://github.com/mrIncompetent))
```
Comment (alvaroaleman) [inline CHANGELOG.md:34]: We should maybe publish the new limits

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,158 @@
...
+) ([mrIncompetent](https://github.com/mrIncompetent))
+- Control plane components are no longer logging at debug level [#2471](https://github.com/kubermatic/kubermatic/issues/2471) ([mrIncompetent](https://github.com/mrIncompetent))
+- Experimantal support for VerticalPodAutoscaler has been added. The VPA resources use the PodUpdatePolicy=initial [#2505](https://github.com/kubermatic/kubermatic/issues/2505) ([mrIncompetent](https://github.com/mrIncompetent))
+- Added 1.11.6 &amp; 1.12.4 to supported Kubernetes versions [#2537](https://github.com/kubermatic/kubermatic/issues/2537) ([mrIncompetent](https://github.com/mrIncompetent))
+- Setting CA bundle with the `oidc-ca-file flag` is now optional [#2562](https://github.com/kubermatic/kubermatic/issues/2562) ([kgroschoff](https://github.com/kgroschoff))
+- Endpoint PUT /api/v1/projects/{project_id} for a renaming of the project is implemented [#2588](https://github.com/kubermatic/kubermatic/issues/2588) ([glower](https://github.com/glower))
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Rename to "Its possible to rename a project"

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,158 @@
...
+- New header params added to delete cluster operation: `DeleteVolumes` and `DeleteLoadBalancers`. Setting `DeleteVolumes` for `true` adds `kubermatic.io/cleanup-in-cluster-pv` finilizer. Setting `DeleteLoadBalancers` for `true` adds `kubermatic.io/cleanup-in-cluster-lb` finilizer. [#2604](https://github.com/kubermatic/kubermatic/issues/2604) ([zreigz](https://
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Replace that with an explanation what the feature is for, no one cares if its done via headers or via something else and no one cares about the names of the finalizers either

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,158 @@
...
+github.com/zreigz))
+- Credentials for Docker Hub are no longer necessary. [#2605](https://github.com/kubermatic/kubermatic/issues/2605) ([kdomanski](https://github.com/kdomanski))
+- Added support for Heptio Ark-based backups [#2617](https://github.com/kubermatic/kubermatic/issues/2617) ([xrstf](https://github.com/xrstf))
```
Comment (alvaroaleman) [inline CHANGELOG.md:52]: Didn't we already have that before @xrstf ? Also a link to the docs on how it works, how it can be setup and how to use it would be nice

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,158 @@
...
+- Running `kubectl get cluster` in a seed now shows some more details [#2622](https://github.com/kubermatic/kubermatic/issues/2622) ([alvaroaleman](https://github.com/alvaroaleman))
+- Added new edpoint: `/api/v1/projects/{project_id}/dc/{dc}/clusters/{cluster_id}/nodedeployments/{nodedeployment_id}/nodes/events` with query parameter `type`. When `?type=warning` then endpoint returns only warning type events. When `?type=normal` then endpoint returns only normal type events. Without query parameters the endpoint returns all events. [#2647]
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Please Remove

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,158 @@
...
+(https://github.com/kubermatic/kubermatic/issues/2647) ([zreigz](https://github.com/zreigz))
+- Kubernetes 1.10 was removed as officially supported version from Kubermatic as its EOL [#2712](https://github.com/kubermatic/kubermatic/issues/2712) ([alvaroaleman](https://github.com/alvaroaleman))
+- Updated machine controller to `v0.10.5` [#2490](https://github.com/kubermatic/kubermatic/issues/2490) ([mrIncompetent](https://github.com/mrIncompetent))
+- Updated dex to 2.12.0 [#2318](https://github.com/kubermatic/kubermatic/issues/2318) ([bashofmann](https://github.com/bashofmann))
+- Updated Ark to 0.10 [requires manual configuration update] [#2615](https://github.com/kubermatic/kubermatic/issues/2615) ([xrstf](https://github.com/xrstf))
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Please clarify? According to the release note it didn't exist before @xrstf

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,158 @@
...
+- [ACTION_REQUIRED] Kubermatic CustomResourceDefinitions have been extracted out of the helm chart. This requires the execution of the `charts/kubermatic/migrate/migrate-kubermatic-chart.sh` script in case the CRD&#39;s where installed without the `&#34;helm.sh/resource-policy&#34;: keep` annotation. [#2459](https://github.com/kubermatic/kubermatic/issues/2459
+) ([mrIncompetent](https://github.com/mrIncompetent))
+- Control plane components are no longer logging at debug level [#2471](https://github.com/kubermatic/kubermatic/issues/2471) ([mrIncompetent](https://github.com/mrIncompetent))
+- Experimantal support for VerticalPodAutoscaler has been added. The VPA resources use the PodUpdatePolicy=initial [#2505](https://github.com/kubermatic/kubermatic/issues/2505) ([mrIncompetent](https://github.com/mrIncompetent))
+- Added 1.11.6 &amp; 1.12.4 to supported Kubernetes versions [#2537](https://github.com/kubermatic/kubermatic/issues/2537) ([mrIncompetent](https://github.com/mrIncompetent))
+- Setting CA bundle with the `oidc-ca-file flag` is now optional [#2562](https://github.com/kubermatic/kubermatic/issues/2562) ([kgroschoff](https://github.com/kgroschoff))
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Please remove, this is part of the "Share cluster" feature and should not have a distinct release note

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,158 @@
...
+- Kubernetes API servers can now be used with OpenID authentication[ACTION REQUIRED] to enable the OpenID for kubernetes API server the users must set `-feature-gates=OpenIDConnectTokens=true` and provide `-oidc-issuer-url`, `-oidc-issuer-client-id` when running the controller. [#2370](https://github.com/kubermatic/kubermatic/issues/2370) ([zreigz](https://git
+hub.com/zreigz))
+- [ACTION REQUIRED] Resource limits for control plane containers have been increased. This might require additional resources for the seed cluster [#2395](https://github.com/kubermatic/kubermatic/issues/2395) ([mrIncompetent](https://github.com/mrIncompetent))
+- Kubermatic controller can now use TLS verification[Action required] to enable the TLS verification for kubermatic-controller-manager the users must set `-feature-gates=OpenIDAuthPlugin=true` and provide `-oidc-issuer-url`, `-oidc-issuer-client-id` and `--oidc-ca-file` when running the controller.[Action required] user need CA bundle file with a correct chain
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Please remove, this is part of the "Share cluster" feature and should not have a distinct release note, also its completely unclear which certificate we validate and why.

I think we need to add a distinct release note about the "share" cluster fature, we seem to have one for every api endpoint added and every flag added and changed, but

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,158 @@
...
+ify`, `client-id` to `oidc-client-id`[Action required] to enable the endpoint the users must set `-feature-gates=OIDCKubeCfgEndpoint=true` and provide `-oidc-client-secret`, `-oidc-redirect-uri` when running the server. If the provider they are working with doesn&#39;t accept `refresh_token` as a scope, setting `-oidc-offline-access-as-scope=false` will pass a
+ppropriate query parameter instead. [#2315](https://github.com/kubermatic/kubermatic/issues/2315) ([p0lyn0mial](https://github.com/p0lyn0mial))
+- [ACTION REQUIRED] kubermatic API command line args were changed: `oidc-issuer-url` to `oidc-url`, `oidc-client-id` to `oidc-authenticator-client-id`, `oidc-offline-access-as-scope` to `oidc-issuer-offline-access-as-scope`, `oidc-redirect-uri` to `oidc-issuer-redirect-uri` and new was added: `oidc-issuer-client-id` [#2338](https://github.com/kubermatic/kuberm
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Please remove, see comment about share cluster feature. AFAIK there is actually no action required

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,158 @@
...
+- [ACTION REQUIRED] added a new command line flag to API server that accepts a set of key=value pairs that enables/disables various features. Existing `enable-prometheus-endpoint` flag is deprecated, the users should use `-feature-gates=PrometheusEndpoint=true` instead.  [#2278](https://github.com/kubermatic/kubermatic/issues/2278) ([p0lyn0mial](https://github
+.com/p0lyn0mial))
+- etcd readiness check timeouts have been increased [#2312](https://github.com/kubermatic/kubermatic/issues/2312) ([mrIncompetent](https://github.com/mrIncompetent))
+- Removed unused fields from cloud specs exposed in the API [#2314](https://github.com/kubermatic/kubermatic/issues/2314) ([maciaszczykm](https://github.com/maciaszczykm))
+- Added a new HTTP endpoint for generating kubeconfig with OIDC tokens. The path of the endpoint is `api/v1/kubeconfig`[Action required] users need to add redirection to api/v1/kubeconfig to dex installation[Action required] kubermatic API command line args were changed:`token-issuer` to `oidc-token-issuer`, `token-issuer-skip-tls-verify` to `oidc-skip-tls-ver
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Please remove, its part of the "share cluster" feature, see comments below

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,158 @@
...
+- Added support for PersistentVolumes on **Hetzner Cloud** [#2613](https://github.com/kubermatic/kubermatic/issues/2613) ([alvaroaleman](https://github.com/alvaroaleman))
+- Openstack Floating IPs will now be de-allocated from your project if they were allocated during node creation [#2675](https://github.com/kubermatic/kubermatic/issues/2675) ([alvaroaleman](https://github.com/alvaroaleman))
+
+
+**Misc:**
+
+- Added support for Kubernetes `v1.13`
+- Kubermatic now supports Kubernetes 1.12 [#2132](https://github.com/kubermatic/kubermatic/issues/2132) ([alvaroaleman](https://github.com/alvaroaleman))
+- The startup time for new clusters was improved [#2148](https://github.com/kubermatic/kubermatic/issues/2148) ([alvaroaleman](https://github.com/alvaroaleman))
+- API endpoints for clusters not separated into projects have been deprecated. [#2244](https://github.com/kubermatic/kubermatic/issues/2244) ([p0lyn0mial](https://github.com/p0lyn0mial))
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Please remove, the API is useless without the Dashboard

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,158 @@
...
+atic/issues/2338) ([zreigz](https://github.com/zreigz))
+- Kubermatic now validates nodes synchronously [#2340](https://github.com/kubermatic/kubermatic/issues/2340) ([alvaroaleman](https://github.com/alvaroaleman))
+- Added endpoints to manage Machine Deployments [#2357](https://github.com/kubermatic/kubermatic/issues/2357) ([maciaszczykm](https://github.com/maciaszczykm))
+- Components will no longer be shown as as unhealthy when only some replicas are up [#2358](https://github.com/kubermatic/kubermatic/issues/2358) ([mrIncompetent](https://github.com/mrIncompetent))
+- Kubernetes API servers can now be used with OpenID authentication[ACTION REQUIRED] to enable the OpenID for kubernetes API server the users must set `-feature-gates=OpenIDConnectTokens=true` and provide `-oidc-issuer-url`, `-oidc-issuer-client-id` when running the controller. [#2370](https://github.com/kubermatic/kubermatic/issues/2370) ([zreigz](https://git
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Please remove and add a note about the "share cluster" feature instead

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,3 +1,158 @@
...
+- Updated machine controller to `v0.10.5` [#2490](https://github.com/kubermatic/kubermatic/issues/2490) ([mrIncompetent](https://github.com/mrIncompetent))
+- Updated dex to 2.12.0 [#2318](https://github.com/kubermatic/kubermatic/issues/2318) ([bashofmann](https://github.com/bashofmann))
+- Updated Ark to 0.10 [requires manual configuration update] [#2615](https://github.com/kubermatic/kubermatic/issues/2615) ([xrstf](https://github.com/xrstf))
+- Updated nginx-ingress-controller to `v0.22.0` [#2668](https://github.com/kubermatic/kubermatic/issues/2668) ([xrstf](https://github.com/xrstf))
+- [ACTION REQUIRED] Updated cert-manager to `v0.6.0` (see https://cert-manager.readthedocs.io/en/latest/admin/upgrading/index.html) [#2674](https://github.com/kubermatic/kubermatic/issues/2674) ([xrstf](https://github.com/xrstf))
+
+
+**Dashboard**:
+- Updated Kubermatic dashboard to `v1.1.0` [#2683](https://github.com/kubermatic/kubermatic/issues/2683) ([mrIncompetent](https://github.com/mrIncompetent))
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Remove, who cares?

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,3 +1,158 @@
...
+    - It is now possible to edit the project name in UI. [#1003](https://github.com/kubermatic/dashboard-v2/issues/1003) ([kgroschoff](https://github.com/kgroschoff))
+    - Machine Networks for VSphere can now be set in the UI [#829](https://github.com/kubermatic/dashboard-v2/issues/829) ([kgroschoff](https://github.com/kgroschoff))
+    - VSphere: Setting a dedicated VSphere user for cloud provider functionalities is now possible. [#834](https://github.com/kubermatic/dashboard-v2/issues/834) ([kgroschoff](https://github.com/kgroschoff))
+    - Fixed that the cluster upgrade link did not appear directly when the details page is loaded [#836](https://github.com/kubermatic/dashboard-v2/issues/836) ([bashofmann](https://github.com/bashofmann))
+    - Kubeconfig can now be shared via a generated link from the UI [#857](https://github.com/kubermatic/dashboard-v2/issues/857) ([kgroschoff](https://github.com/kgroschoff))
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Extend that to include a link to the docs and remove all other "share cluster"-related changelog entries?

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,3 +1,158 @@
...
+    - Fixed duplicated SSH keys in summary view during cluster creation. [#879](https://github.com/kubermatic/dashboard-v2/issues/879) ([kgroschoff](https://github.com/kgroschoff))
+    - On project change, the user will stay on the same page, if he has the corresponding rights. [#889](https://github.com/kubermatic/dashboard-v2/issues/889) ([kgroschoff](https://github.com/kgroschoff))
+    - Fixed issues with caching the main page. [#893](https://github.com/kubermatic/dashboard-v2/issues/893) ([maciaszczykm](https://github.com/maciaszczykm))
+    - Added support for creating, viewing, updating and deleting node deployments. [#949](https://github.com/kubermatic/dashboard-v2/issues/949) ([maciaszczykm](https://github.com/maciaszczykm))
```
Comment (alvaroaleman) [inline CHANGELOG.md]: Just keep this and remove all other occurences of "NodeDeployment" stuff?

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,3 +1,158 @@
...
+- Updated Minio to `RELEASE.2019-01-16T21-44-08Z` [#2667](https://github.com/kubermatic/kubermatic/issues/2667) ([xrstf](https://github.com/xrstf))
+- metrics-server will use 2 replicas [#2707](https://github.com/kubermatic/kubermatic/issues/2707) ([mrIncompetent](https://github.com/mrIncompetent))
+
+
+**Security:**
+
+- The admin token can no longer be read through the Kubermatic API. [#2105](https://github.com/kubermatic/kubermatic/issues/2105) ([p0lyn0mial](https://github.com/p0lyn0mial))
+- Communicating with cloud providers through the project APIs no longer requires providing additional credentials. [#2180](https://github.com/kubermatic/kubermatic/issues/2180) ([p0lyn0mial](https://github.com/p0lyn0mial))
+- [ACTION REQUIRED] to enable the Double Submit Cookies solution the users must set `-feature-gates=OIDCKubeCfgEndpoint=true` and provide settings for cookie `oidc-issuer-cookie-hash-key` and`oidc-issuer-cookie-secure-mode`  [#2360](https://github.com/kubermatic/kubermatic/issues/2360) ([zreigz](https://github.com/zreigz))
```
Comment (alvaroaleman) [inline CHANGELOG.md]: I have no idea what the `Double Submit Cookies solution` is

File: CHANGELOG.md [hunk 18]
```diff
@@ -1,3 +1,158 @@
...
+- cloud-config files are now properly escaped [#2498](https://github.com/kubermatic/kubermatic/issues/2498) ([alvaroaleman](https://github.com/alvaroaleman))
+- SSH keys can no longer be added with duplicate names [#2499](https://github.com/kubermatic/kubermatic/issues/2499) ([kgroschoff](https://github.com/kgroschoff))
+- Fixed an issue with kubelets being unreachable by the apiserver on some OS configurations. [#2522](https://github.com/kubermatic/kubermatic/issues/2522) ([mrIncompetent](https://github.com/mrIncompetent))
+- Timestamp format has been unified throughout the Kubermatic API. [#2534](https://github.com/kubermatic/kubermatic/issues/2534) ([zreigz](https://github.com/zreigz))
+- Updated cert-manager to fix an issue which caused re-issuing of a certficate via the http01 challenge to fail [#2658](https://github.com/kubermatic/kubermatic/issues/2658) ([alvaroaleman](https://github.com/alvaroaleman))
+- NodeDeployments can no longer be configured to provision kubelets at versions incompatible with the control plane. [#2665](https://github.com/kubermatic/kubermatic/issues/2665) ([kdomanski](https://github.com/kdomanski))
```
Comment (alvaroaleman) [inline CHANGELOG.md]: I'd remove that and just stay with one release node regarding `NodeDeployments`

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,3 +1,158 @@
...
+
+
+
+
+### [v2.8.6]()
+
+- Added support for Kubernetes `v1.13` [#2628](https://github.com/kubermatic/kubermatic/issues/2628) ([mrIncompetent](https://github.com/mrIncompetent))
+- Fixed reconciling of deep objects [#2630](https://github.com/kubermatic/kubermatic/issues/2630) ([mrIncompetent](https://github.com/mrIncompetent))
```
Comment (alvaroaleman) [inline CHANGELOG.md:133]: Maybe just write `Fixed reconciling`? I doubt many customers know what a `deep object` is

**Comment by kdomanski** [comment]
@alvaroaleman care to take another look?

File: CHANGELOG.md (hunk 12 above)
Comment (alvaroaleman) [reply CHANGELOG.md]: ?

File: CHANGELOG.md (hunk 10 above)
Comment (alvaroaleman) [reply CHANGELOG.md]: ?

File: CHANGELOG.md (hunk 9 above)
Comment (alvaroaleman) [reply CHANGELOG.md]: ?

File: CHANGELOG.md (hunk 8 above)
Comment (alvaroaleman) [reply CHANGELOG.md]: ?

File: CHANGELOG.md (hunk 5 above)
Comment (alvaroaleman) [reply CHANGELOG.md:52]: ?

File: CHANGELOG.md (hunk 7 above)
Comment (alvaroaleman) [reply CHANGELOG.md]: ?

File: CHANGELOG.md (hunk 14 above)
Comment (alvaroaleman) [reply CHANGELOG.md]: ?

File: CHANGELOG.md (hunk 16 above)
Comment (alvaroaleman) [reply CHANGELOG.md]: Maybe reword to something like "Nodes are now being managed as NodeGroups, this allows to easely change settings for a group of Nodes"?

File: CHANGELOG.md (hunk 5 above)
Comment (kdomanski) [reply CHANGELOG.md:52]: IMO this is just chart sync being fixed, I see ark charts as old as v2.8.0-rc1. cc @xrstf

Comment (xrstf) [reply CHANGELOG.md:52]: I see no ark/ark-config charts in https://github.com/kubermatic/kubermatic-installer/tree/release/v2.8/charts and the sync-script in 2.8 also does not contain anything related to ark (https://github.com/kubermatic/kubermatic/blob/release/v2.8/api/hack/sync-charts.sh).

Comment (kdomanski) [reply CHANGELOG.md:52]: Yes, that's my point exactly. There's nothing about ark in the installer repo BECAUSE the sync is not there, but there are charts in the main Kubermatic repo.

Comment (xrstf) [reply CHANGELOG.md:52]: Well it's in the changelog because we added it in 2.9:

* https://github.com/kubermatic/kubermatic/blob/release/v2.9/api/hack/sync-charts.sh#L27
* https://github.com/kubermatic/kubermatic-installer/tree/release/v2.9/charts

I don't understand the issue.

File: CHANGELOG.md (hunk 19 above)
Comment (kdomanski) [reply CHANGELOG.md:133]: That's okay that many don't know. I don't want to suggest that all reconcilling was broken.
